src/Couchbase.Lite.Shared/API/Database/Collection.cs

```diff
@@ -269,13 +321,108 @@ public void DeleteIndex([NotNull] string name)
 
         #endregion
 
-        #region IDispose
+        #region Public Methods - QueryFactory
 
-        public void Dispose()
+        public IQuery CreateQuery([NotNull] string queryExpression)
         {
             throw new NotImplementedException();
         }
 
         #endregion
+
+        #region Internal Methods
+
+        internal void GetSpec()
+        {
+            ThreadSafety.DoLocked(() =>
+            {
+                if (c4coll == null)
+                    return;
+
+                var spec = Native.c4coll_getSpec(c4coll);
+                Name = spec.name.CreateString();
+                Scope.Name = spec.scope.CreateString();
+            });
+        }
+
+        /// <summary>
+        /// Returns false if this collection has been deleted, or its database closed.
+        /// </summary>
+        internal bool IsCollectionValid()
+        {
+            bool isValid = false;
+            ThreadSafety.DoLocked(() =>
+            {
+                isValid = Native.c4coll_isValid(_c4coll);
+                if (!isValid) {
+                    isValid = Native.c4coll_isValid(_c4coll);
+                }
+            });
+
+            return isValid;
+        }
+
+        #endregion
+
+        #region Private Methods
+
+        private C4Database* GetC4Database()
+        {
+            C4Database* c4db = null;
+            ThreadSafety.DoLocked(() =>
+            {
+                if (c4coll == null)
+                    return;
+
+                c4db = Native.c4coll_getDatabase(c4coll);
+            });
+
+            return c4db;
+        }
+
+        private void ReleaseCollection()
+        {
+            Native.c4coll_release(_c4coll);
+            _c4coll = null;
```

Comment:
Depending on where this is called this could open a race.  Prefer safe swapping the member first, and then releasing it.

Reply:
Done.